### PR TITLE
ICU-21419 Fix memory leak in AliasReplacer

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21419
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [X] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/common/locid.cpp [hunk 1]
```diff
@@ -1422,20 +1422,22 @@ AliasReplacer::replaceTerritory(UVector& toBeFreed, UErrorCode& status)
...
                          foundInReplacement[len] == '\0');
-                item = new CharString(foundInReplacement, (int32_t)len, status);
+                item.adoptInsteadAndCheckErrorCode(
+                    new CharString(foundInReplacement, (int32_t)len, status), status);
             }
         }
-        if (item == nullptr) {
-            item = new CharString(replacement,
-                                  (int32_t)(firstSpace - replacement), status);
+        if (item.isNull()) {
```
Comment (markusicu) [inline icu4c/source/common/locid.cpp:1429]:
```suggestion
        if (item.isNull() && U_SUCCESS(status)) {
```
We should not try to create a new CharString if something went wrong.

I suggest using `git commit --amend` to keep it to one commit.

Comment (FrankYFTang) [reply icu4c/source/common/locid.cpp:1429]: DONE

Comment (FrankYFTang) [reply icu4c/source/common/locid.cpp:1429]: PTAL

Comment (markusicu) [reply icu4c/source/common/locid.cpp:1429]: Sorry, looking at GitHub I don't see this line changing. What am I missing?

Comment (FrankYFTang) [reply icu4c/source/common/locid.cpp:1429]: git commit --amend didn't work as I expected.

**Review by markusicu — APPROVED** [review]
lgtm

**Pull request merged**